### PR TITLE
feat: add proof keeper

### PR DESCRIPTION
**Pull request by will-2012**
### Description

Add the proof keeper(only in path-base, bufferlist config) component to store the proposed proof in the background, and the op-proposal can query the proof in the proof keeper.

### Rationale

nodebufferlist watch the block height and stores the proof of the corresponding block height according to the configuration.

### Example

```shell
./op-geth \
  ...
  --state.scheme path \
  --pathdb.nodebuffer list \
  --pathdb.proposeblock 240 \
  --pathdb.enableproofkeeper \
  --pathdb.keepproofblockspan 100000 \ # This is optional, default is 90000.
  ...
``` 
This `pathdb.enableproofkeeper` enable proof keeper.

### Changes

Notable changes:
* core: add proof keeper skeleton.
* rawdb: add keeper meta and proof data api.
* pathdb: add watcher func in nodebufferlist.
* API: adjust get proof API, can dispatch request to proof keeper.

### CODE REVIEW

File: cmd/utils/flags.go [hunk 1]
```diff
@@ -35,6 +34,8 @@ import (
 	"strings"
 	"time"
 
+	"github.com/ethereum/go-ethereum/core/opcodeCompiler/compiler"
+
```
Comment (sysvm) [inline cmd/utils/flags.go:38]: Remove the extra blank line.

File: cmd/utils/flags.go [hunk 2]
```diff
@@ -313,6 +314,18 @@ var (
 		Value:    pathdb.DefaultProposeBlockInterval,
 		Category: flags.StateCategory,
 	}
+	EnableProofKeeperFlag = &cli.BoolFlag{
+		Name:     "pathdb.enableproofkeeper",
+		Usage:    "Enable path db proof keeper for store proposed proof",
```
Comment (sysvm) [inline cmd/utils/flags.go:319]: `store` -> `storing`

File: cmd/utils/flags.go (hunk 1 above)
Comment (will-2012) [reply cmd/utils/flags.go]: Updated.

File: cmd/utils/flags.go (hunk 2 above)
Comment (will-2012) [reply cmd/utils/flags.go]: Updated.

**Review by krish-nr — APPROVED** [review]
LGTM

**Review by sysvm — DISMISSED** [review]

**Review by welkin22 — APPROVED** [review]
LGTM

**Review by redhdx — APPROVED** [review]
LGTM

**Pull request merged**